### PR TITLE
Add new MCode to reset bed Z height

### PR DESCRIPTION
**Pull request by jminardi**
This PR adds a new MCode M852 that will adjust the zprobe offset such that the current height is set to 0. There is an optional argument `Z` that allows you to assert you are in a different height above the bed.

This PR also updates the test script to connect and disconnect between each test method.

### CODE REVIEW

**Comment by jminardi** [comment]
CC @pizzyflavin @kdumontnu

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -5952,6 +5954,29 @@ inline void gcode_M503() {
...
+* Optional arguments:
+*
+*   Z - How high off the bed in mm you are asserting the hotend is.
+*   V - Verbose mode (echo the new zprobe offset)
+*/
+inline void gcode_M852() {
+  // additional_offset is how high off the bed you are asserting you are
+  float additional_offset = code_seen('Z') ? code_value() : 0;
+  zprobe_zoffset += (current_position[Z_AXIS] - additional_offset);
+  current_position[Z_AXIS] = 0;
```
Comment (pizzyflavin) [inline Marlin/Marlin_main.cpp]: Shouldn't this be `current_position[Z_AXIS] = additional_offset` ?

Comment (jminardi) [inline Marlin/Marlin_main.cpp]: You are correct, fixed.

**Comment by pizzyflavin** [comment]
LGTM

**Pull request merged**